src/main/scala/uk/gov/nationalarchives/dp/client/Client.scala

```diff
@@ -0,0 +1,139 @@
+package uk.gov.nationalarchives.dp.client
+
+import cats.MonadError
+import cats.implicits._
+import sttp.capabilities.Streams
+import sttp.client3._
+import sttp.client3.upicklejson._
+import uk.gov.nationalarchives.dp.client.Client.{AuthDetails, BitStreamInfo, Entity}
+import upickle.default._
+
+import java.time.format.DateTimeFormatter
+import java.time.ZonedDateTime
+import java.util.UUID
+import scala.xml.{Elem, XML}
+
+trait Client[F[_], S] {
+  def metadataForEntityUrl(url: String, auth: AuthDetails): F[Seq[Elem]]
```

Comment:
Do you think it's worth abstracting the Preservica data model here; taking an id like `io:{uuid}` and inferring the type/uuid instead of passing URLs outside of the library?